common/src/main/java/com/gregtechceu/gtceu/data/pack/GTDynamicResourcePack.java

```diff
@@ -0,0 +1,208 @@
+        this.name = name;
+        CLIENT_DOMAINS.addAll(domains);
+    }
+
+    public static void clearClient() {
+        DATA.clear();
+    }
+
+    public static void addBlockModel(ResourceLocation loc, JsonElement obj) {
+        ResourceLocation l = getModelLocation(loc);
+        if (ConfigHolder.INSTANCE.dev.dumpAssets) {
+            Path parent = Platform.getGamePath().resolve("gtceu/dumped/assets");
+            writeJson(l, null, parent, obj);
+        }
+        DATA.put(l, obj.toString().getBytes(StandardCharsets.UTF_8));
+    }
+
+    public static void addBlockModel(ResourceLocation loc, Supplier<JsonElement> obj) {
+        addBlockModel(loc, obj.get());
+    }
+
+    public static void addItemModel(ResourceLocation loc, JsonElement obj) {
+        ResourceLocation l = getItemModelLocation(loc);
+        if (ConfigHolder.INSTANCE.dev.dumpAssets) {
+            Path parent = Platform.getGamePath().resolve("gtceu/dumped/assets");
+            writeJson(l, null, parent, obj);
+        }
+        DATA.put(l, obj.toString().getBytes(StandardCharsets.UTF_8));
+    }
+
+    public static void addItemModel(ResourceLocation loc, Supplier<JsonElement> obj) {
+        addItemModel(loc, obj.get());
+    }
+
+    public static void addBlockState(ResourceLocation loc, JsonElement stateJson) {
+        ResourceLocation l = getBlockStateLocation(loc);
+        if (ConfigHolder.INSTANCE.dev.dumpAssets) {
+            Path parent = Platform.getGamePath().resolve("gtceu/dumped/assets");
+            writeJson(l, null, parent, stateJson);
+        }
+        DATA.put(l, stateJson.toString().getBytes(StandardCharsets.UTF_8));
+    }
+
+    public static void addBlockState(ResourceLocation loc, Supplier<JsonElement> generator) {
+        addBlockState(loc, generator.get());
+    }
+
+    public static void addBlockTexture(ResourceLocation loc, byte[] data) {
+        ResourceLocation l = getTextureLocation("block", loc);
+        if (ConfigHolder.INSTANCE.dev.dumpAssets) {
+            Path parent = Platform.getGamePath().resolve("gtceu/dumped/assets");
+            writeByteArray(l, null, parent, data);
+        }
+        DATA.put(l, data);
+    }
+
+    public static void addItemTexture(ResourceLocation loc, byte[] data) {
+        ResourceLocation l = getTextureLocation("item", loc);
+        if (ConfigHolder.INSTANCE.dev.dumpAssets) {
+            Path parent = Platform.getGamePath().resolve("gtceu/dumped/assets");
+            writeByteArray(l, null, parent, data);
+        }
+        DATA.put(l, data);
+    }
+
+    @ApiStatus.Internal
+    public static void writeByteArray(ResourceLocation id, @Nullable String subdir, Path parent, byte[] data) {
+        try {
+            Path file;
+            if (subdir != null) {
+                file = parent.resolve(id.getNamespace()).resolve(subdir).resolve(id.getPath() + ".png"); // assume PNG
+            } else {
+                file = parent.resolve(id.getNamespace()).resolve(id.getPath());
```

Comment:
Should this have `+ ".png"` as well?

Reply:
no, read the comment above the `if`

Reply:
There is no comment above the if

Reply:
damn where'd my comment go
 // assume the file type is also appended if the full path is set.